src/binary/merkle_tree.rs

```diff
@@ -98,7 +98,10 @@ where
         let root_node = self.root_node()?.unwrap();
```

Comment:
I know it's not related to this PR so feel free to ignore but are these unwraps safe / meant to be in production code?

Reply:
This line is from the `prove` method; if you've reached this line in `prove`, you are guaranteed to have a root node. This is guaranteed because:
- A BMT with only one node (leaf) has that one node as its root node. A BMT with zero leaves has no root node.
- `prove` requires a proof index (`u64`). The provided index must be less than the leaves count; if you have zero leaves, you cannot provide a valid index, and the method will early exit with an `Err`

I can add some inline safety notes to the code to help readers understand the use `unwrap` here.

Reply:
Yeh maybe an expect with a quick summary of why it's unreacchable